src/inspector_profiler.cc

```diff
@@ -218,6 +224,48 @@ void V8CoverageConnection::End() {
   CHECK_EQ(ending_, false);
   ending_ = true;
   DispatchMessage("Profiler.takePreciseCoverage");
+  WriteSourceMapCache();
+}
+
+void V8CoverageConnection::WriteSourceMapCache() {
+  Isolate* isolate = env_->isolate();
+  Local<Context> context = env_->context();
+  HandleScope handle_scope(isolate);
+  Context::Scope context_scope(context);
+
+  Local<Object> process_object = env_->process_object();
+  Local<Value> get_source_map_cache_fn = process_object->Get(env()->context(),
+    FIXED_ONE_BYTE_STRING(isolate, "_getSourceMapCache"))
+    .ToLocalChecked();
+
+  Local<Value> source_map_cache_v;
+  Local<Value> argv[0] = {};
+  if (!get_source_map_cache_fn.As<Function>()->Call(
+       env()->context(), process_object, 0, argv)
+      .ToLocal(&source_map_cache_v)) {
+    return;
+  }
+
+  Local<Object> source_map_cache = source_map_cache_v.As<Object>();
+
+  Local<String> result_s;
+  if (!v8::JSON::Stringify(context, source_map_cache).ToLocal(&result_s)) {
+    fprintf(stderr, "Failed to stringify source-map cache");
+    return;
+  }
+
+  // Create the directory if necessary.
+  std::string directory = GetDirectory();
+  DCHECK(!directory.empty());
+  if (!EnsureDirectory(directory, type())) {
+    return;
+  }
+
+  std::string filename = GetFilename("source-map");
+  DCHECK(!filename.empty());
+  std::string path = directory + kPathSeparator + filename;
+
+  WriteResult(env_, path.c_str(), result_s);
```

Comment:
I guess I'm still not 100% sure why writing the cache file is necessary.

Reply:
similarly to coverage, the problem we run into is that a process might spawn several subprocesses (as an example tap runs tests in subprocesses). In these scenarios it becomes difficult for top level tooling to capture source maps, coverage, etc., from the subprocesses. 

What tools like `nyc` did was install a shim node binary, which injected `process.on('exit')` hooks, and signal handlers ... but this is fairly error prone (and a little bit terrifying).

I think the ideal would be a programmatic API that gives a user access to the source map information in the running process, along with functionality similar to coverage, where we can just dump the information for upstream tooling.

Reply:
@jasnell I wonder if we should perhaps introduce a flag similar to `NODE_V8_COVERAGE` however, and decouple writing the information to disk from the source-map cache functionality ... how do we think someone should enable the `source-map-cache`? perhaps:

```
--enable-experimental-source-maps
```

and if you happen to also have NODE_V8_COVERAGE set, we write the source-maps to disk?

Reply:
That could work

Reply:
Making source-maps write to disk if `NODE_V8_COVERAGE` is enabled seems like the right call  (otherwise coverage can't be properly collected for code executed by tools like `ts-node` and `@babel/node`, which transpile on the fly and never output content to disk).

I think it's reasonable to keep the `source-map` API private for now, with the only interface being through `NODE_V8_COVERAGE`, and then in future work we can begin to:

1. expose a public API.
2. start integrating source-maps more tightly with V8/error objects.

Reply:
(basically I'm advocating not introducing `--enable-experimental-source-maps` until we introduce APIs that run independently of `NODE_V8_COVERAGE`).